build-support/docker/centos6/Dockerfile

```diff
@@ -4,18 +4,40 @@
 # Use centos6 for compatibility with glibc 2.12.
 FROM centos:6
 
-# Install python 2.7.13, (more) modern gcc, and a JDK.
+# Install a (more) modern gcc, and a JDK.
+# Also install sqlite-devel because python 3 errors on startup if not -- see
+# https://stackoverflow.com/a/39907500/2518889.
 RUN yum -y update
+# TODO: figure out why this needs to be installed first for /usr/bin/scl to work!
 RUN yum install -y centos-release-scl
 RUN yum install -y \
-  devtoolset-7-gcc \
-  devtoolset-7-gcc-c++ \
-  git \
-  java-1.8.0-openjdk-devel \
-  python27 \
-  libffi \
-  libffi-devel \
-  openssl-devel
+        bzip2{,-devel} \
+        devtoolset-7-gcc{,-c++} \
+        git \
+        java-1.8.0-openjdk-devel \
+        libffi{,-devel} \
+        openssl-devel \
+        readline{,-devel} \
+        sqlite-devel \
+        zlib{,-devel}
 
-# By default, execute in an environment with python27 enabled.
-ENTRYPOINT ["/usr/bin/scl", "enable", "python27", "devtoolset-7",  "--"]
+# Install the desired versions of python 2.7 and 3.6.
+ARG PYTHON_27_VERSION=2.7.13
+ARG PYTHON_36_VERSION=3.6.8
+# NB: PYENV_ROOT must be set for `pyenv install` to be available. This failure mode is not mentioned
+# in https://github.com/pyenv/pyenv#basic-github-checkout.
+ENV PYENV_ROOT /pyenv-docker-build
```

Comment:
For parity with our UCS2 shard and the new bash script, I'd move `ENV PYENV_BIN` to be right underneath this line. `PYENV_BIN` doesn't have to actually exist yet—it's only a path to where it will be.